### ISSUE TITLE
Configurable SMS OTP Timeout

### ISSUE THREAD

**Issue by mdp18** [nhost/nhost#856]
I'd like to request a feature that allows app developers the ability to configure the timeout of an OTP for passwordless and mfa.

**Comment by elitan** [comment]
Do you mean that an OTP should only be valid for a certain time? Say 5 minutes? And you want that be able to configure that "5-minute" value?

**Comment by jbergstroem** [comment]
> Do you mean that an OTP should only be valid for a certain time? Say 5 minutes? And you want that be able to configure that "5-minute" value?

I would like to see this as well - a configurable timeout for when the OTP should expire.

I'm guessing this is set here at the moment?

https://github.com/nhost/hasura-auth/blob/a8941340623eebb533801b3de87e79589ba936e5/src/utils/otp.ts#L5-L9

**Comment by mdp18** [comment]
> Do you mean that an OTP should only be valid for a certain time? Say 5 minutes? And you want that be able to configure that "5-minute" value?

Yes thats right so if I wanted to configure the OTP to expire in 3 minutes, I would be able to do so. I'd mainly like to have this capability globally.
